tests/functional/codegen/calling_convention/test_erc20_abi.py

```diff
@@ -90,13 +90,12 @@ def test_call_transfer(w3, erc20, erc20_caller, assert_tx_failed):
     assert erc20.balanceOf(w3.eth.accounts[1]) == 10
 
     # more than allowed
-    assert_tx_failed(lambda: erc20_caller.transfer(w3.eth.accounts[1], TOKEN_TOTAL_SUPPLY))
+    with assert_tx_failed():
+        erc20_caller.transfer(w3.eth.accounts[1], TOKEN_TOTAL_SUPPLY)
 
     # Negative transfer value.
-    assert_tx_failed(
-        function_to_test=lambda: erc20_caller.transfer(w3.eth.accounts[1], -1),
-        exception=ValidationError,
-    )
+    with assert_tx_failed(ValidationError):
```

Comment:
this is kind of an interesting use of `assert_tx_failed` because it exposes a naming issue with `assert_tx_failed` -- `assert_tx_failed` doesn't actually necessarily check for transaction failure, it could test any exception (and just anchors the state)